R/stan_module.R

```diff
@@ -21,6 +21,17 @@ StanModule <- setClass(
 
 )
 
+#' @importFrom assertthat is.scalar
+setValidity("StanModule", function(object) {
+    msg <- NULL
+
+    if (!is.scalar(object@priors) && is.null(names(object@priors))) {
+        msg <- c(msg, "`Priors` must have names")
+    }
+
+    msg
+})
+
 
```

Comment:
I believe those check errors are from R codes here, and almost all of them reported the same error info as follows
![image](https://user-images.githubusercontent.com/56521967/194215709-539fcac7-c6b7-440f-8029-116bfcf943b2.png)

Reply:
If these functions were coded by assertion, it will be fine. 
`if (length(priors) > 0) {`
` assert_that(`
   `is.null(names(priors)) == FALSE,`
       ` all(names(priors) != ""),`
       ` msg = "Priors must have names"`
`    )`
`}` However, I cannot really understand the functional difference between my new code and the assertion code before.

Reply:
So when I am looking at the `R CMD check` errors, most of them come from an object where the priors does not have names. I think you can start from those tests/errors and see where we are missing the prior names, and then we add them there?
@gowerc fyi